.github/workflows/main.yml

```diff
@@ -89,6 +89,10 @@ jobs:
         cd forte-wrappers
         pip install src/spacy
 
+    - name: Install NeuralCoref
```

Comment:
I don't think we should add installation of NeuralCoref to the workflow directly. @hunterhector what do you think? Should we maybe add an extras_require in setup.py like we do in forte-wrappers? Not sure if that will make sure installation happens after space and Cython though.

Reply:
Yeah, if this is a feature we should include them in the `setup.py`(and yes, as an extras_require). Otherwise the user won't be able to use the module.

However, not only `NeuralCoref`, right now I saw `forte.spacy` is also installed in the workflows only.

The correct setting should be:
1. For each functionality (say Neural Coref or Scispacy), they need to have their own `extra_require` session.
2. To run each module, only the relevant module needs to be installed, thus no dependency issue. During test, these different test scenarios need to be defined via the job matrix (example [1](https://github.com/asyml/forte/blob/master/.github/workflows/main.yml#L141) [2](https://github.com/asyml/forte-wrappers/blob/main/.github/workflows/main.yml#L79) 
3. To run modules in one single pipeline, right now we don't have a very easy way to do it (it is being worked on). We can use `remote_processor` for now.

Reply:
@Piyush13y a side note for you, the `required` install [section](https://github.com/asyml/ForteHealth/blob/master/setup.py#L24) is redundant. This looks like a copy from the `Forte` requirements, which isn't needed as long as you include `forte` as the requirement. A copy will actually create potential conflicts.

Reply:
I found two solutions to install neuralcoref's dependencies before neuralcoref.
1. use 'setup_requires' keyword. This works fine with `python setup.py install`, but with `pip install` it does not work, because pip does not support this keyword. Moreover, the document says this keyword is deprecated, so I decided to use another plan.
2. Another plan is a bit nasty, but I can't find a better one. I add a `subprocess.call` to pip install the dependencies before `setuptools.setup`.